### PR TITLE
lint: fix spelling to American english

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: #6030

### CODE REVIEW

**Review by alexanderbez — APPROVED** [review]
ACK -- curious, why is this breaking?

**Comment by tessr** [comment]
I think because it renames the "behaviour" package. Maybe we should walk all of this back for now; I did not mean to trigger all of this with the linter addition

**Comment by alexanderbez** [comment]
Ahhh, right! That's pretty unfortunate. Well do we rip the bandaid off now or just never do it?

**Comment by cmwaters** [comment]
I believe the behaviour package will most likely be made redundant by the p2p refactor which should define it's own way of handling peers. Also this package is currently only used by the v2 blockchain reactor

**Comment by tessr** [comment]
If it's only used by the blockchain v2 reactor, maybe it's OK to rip the bandaid off...

**Comment by cmwaters** [comment]
It's all internal (or at least should be) to v2 so maybe we could actually make it internal and move the behavior stuff into the v2 reactor. Then at a later point if we decide to integrate other reactors with the peer manager we can slide out the behavior stuff for the new system

**Comment by tessr** [comment]
Okay. Down the hatch! Let's do it!

**Comment by tessr** [comment]
I think this merits a changelog entry as well.

File: blockchain/v2/reporter.go [hunk 1]
```diff
@@ -7,13 +7,13 @@ import (
 	"github.com/tendermint/tendermint/p2p"
 )
 
-// Reporter provides an interface for reactors to report the behaviour
+// Reporter provides an interface for reactors to report the behavior
 // of peers synchronously to other components.
 type Reporter interface {
```
Comment (cmwaters) [inline blockchain/v2/reporter.go:12]: These are still publicly exposed interfaces. Should I make them private?

Comment (alexanderbez) [reply blockchain/v2/reporter.go:12]: If they're moved into internal, it won't matter I believe.

Comment (cmwaters) [reply blockchain/v2/reporter.go:12]: ah right but it isn't in an `internal` folder. I've just moved it to be a part of the blockchain/v2 package

Comment (alexanderbez) [reply blockchain/v2/reporter.go:12]: It seems we have two options (assuming they're only used within blockchain v2):

1. Use an `internal` package and keep it exported
2. Make these types private

Comment (cmwaters) [reply blockchain/v2/reporter.go:12]: Yup. What's your preference?

Comment (alexanderbez) [reply blockchain/v2/reporter.go:12]: I have none, but keep in mind:

Option (1):
- leans towards the approach we're ultimately going to take towards Tendermint 1.0, so it's more "idiomatic"
- the PR diff will be larger, but I guess this isn't a big deal

Option (2):
- probably less work for you
- smaller PR diff

Comment (tessr) [reply blockchain/v2/reporter.go:12]: Let's do Option 1, but as a separate PR from all these spelling changes. I can revert the linter change if it's causing trouble between now and then.

Comment (cmwaters) [reply blockchain/v2/reporter.go:12]: Cool shall I revert the last couple of commits and we get this in as the lint pleaser PR and then I will open a new one internalizing the behavior package

Comment (tessr) [reply blockchain/v2/reporter.go]: Yep. I'll remove the linter in a separate PR as well

Comment (cmwaters) [reply blockchain/v2/reporter.go]: I don't think you will need to

**Review by melekes — APPROVED** [review]

**Comment by cmwaters** [comment]
So apart from the behavior package the only "code" thing we are changing is renaming the `Subscription` method in libs/pubsub/ from `Cancelled` to `Canceled`. This shouldn't be a problem (famous last words)

**Comment by tessr** [comment]
@cmwaters can you also add a changelog pending entry for the renamed code?

**Comment by cmwaters** [comment]
Done it here: https://github.com/tendermint/tendermint/pull/6100

**Pull request merged**